src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -64,12 +65,13 @@ private abstract partial class CSharpCodeGenerator : CodeGenerator<StatementSynt
             protected CSharpCodeGenerator(
                 InsertionPoint insertionPoint,
                 SelectionResult selectionResult,
-                AnalyzerResult analyzerResult)
-                : base(insertionPoint, selectionResult, analyzerResult)
+                AnalyzerResult analyzerResult,
+                bool extractLocalFunction)
```

Comment:
you inconsitently use extractLocalFunctoin and generateLocalFunction.  'extraction/generation' is implied by the feature.  so just consider having this be `bool localFunction` everywhere. Alternatively allow caller to pass in `MethodKind`. (though the boolean approach is fine with me).